common/transform/src/main/java/ai/h2o/mojos/deploy/common/transform/MojoScorer.java

```diff
@@ -96,10 +94,10 @@ public ScoreResponse score(ScoreRequest request) {
       ShapleyType requestedShapleyType = shapleyType(request.getRequestShapleyValueType());
       switch (requestedShapleyType) {
         case TRANSFORMED:
-          response.setFeatureShapleyContributions(transformedFeatureContribution(request));
+          response.setFeatureShapleyContributions(featureContribution(request, false));
           break;
         case ORIGINAL:
-          log.info(UNIMPLEMENTED_MESSAGE);
+          response.setFeatureShapleyContributions(featureContribution(request, true));
```

Comment:
I think we need to know what other DAI models doesn't support Shap values. Is there anything apart from regression/binomial/multinomial models? Also how do we handle older DAI models that doesn't have the original matrix. Can we throw error to the score endpoint if requested with Shap? @mmalohlava might need to help us on this.

Reply:
> Is there anything apart from regression/binomial/multinomial models?

For DAI MOJOs, for sure.

---

IMHO, having to take stock of the full set of allowed/disallowed set of DAI and H2O-3 models, instead of failing-fast when a dependent library fails, may suggest that we're doing something in a particularly difficult-to-maintain way.

There are numerous ways for scoring requests to fail (for both DAI and H2O-3 MOJOs) once the scorer hands the task off to the underlying MOJO2 lib.  Even in the case of Shap scoring, I imagine there are cases wherein users might need to be made aware that something went wrong, since the issue wasn't a lack of support, but rather a legitimate error.